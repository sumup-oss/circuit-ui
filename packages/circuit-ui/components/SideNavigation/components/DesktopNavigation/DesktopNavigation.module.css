.wrapper {
  --primary-navigation-width: 48px;
  --primary-navigation-width-open: 220px;
}

@media (max-width: 1279px) {
  .wrapper {
    display: none;
  }
}

@media (min-width: 1280px) {
  .wrapper {
    flex-shrink: 0;
    min-width: var(--primary-navigation-width);
  }
}

@media (min-width: 1900px) {
  .wrapper {
    min-width: var(--primary-navigation-width-open);
  }
}

.primary {
  position: fixed;
  top: var(--top-navigation-height, 0);
  bottom: 0;
  left: 0;
  z-index: var(--cui-z-index-navigation);
  display: flex;
  flex-direction: column;
  width: var(--cui-icon-sizes-tera);
  height: calc(100vh - var(--top-navigation-height, 0px));
  padding-top: var(--cui-spacings-kilo);
  overflow-x: hidden;
  overflow-y: auto;
  background-color: var(--cui-bg-normal);
  box-shadow: 1px 0 var(--cui-border-divider);
  transition:
    width var(--cui-transitions-default),
    box-shadow var(--cui-transitions-default);
}

.primary:hover,
.primary:focus-within {
  width: var(--primary-navigation-width-open);
  box-shadow: 0 3px 8px 0 rgb(0 0 0 / 20%);
}

@media (min-width: 1900px) {
  .primary {
    width: var(--primary-navigation-width-open);
  }

  .primary:hover,
  .primary:focus-within {
    box-shadow: 1px 0 var(--cui-border-divider);
  }
}

.secondary {
  position: sticky;
  top: var(--top-navigation-height, 0);
  display: flex;
  flex-direction: column;
  width: 200px;
  height: calc(100vh - var(--top-navigation-height, 0px));
  margin-left: var(--primary-navigation-width);
  overflow-y: auto;
  background-color: var(--cui-bg-normal);
  border-right: var(--cui-border-width-kilo) solid var(--cui-border-divider);
}

@media only screen and (min-width: 1900px) {
  .secondary {
    margin-left: var(--primary-navigation-width-open);
  }
}

.list {
  padding: 0;
  margin: 0;
  list-style: none;
}

.headline {
  margin: var(--cui-spacings-giga) var(--cui-spacings-mega)
    var(--cui-spacings-kilo);
}
