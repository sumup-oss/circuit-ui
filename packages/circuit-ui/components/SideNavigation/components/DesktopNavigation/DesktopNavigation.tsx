/**
 * Copyright 2021, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* eslint-disable jsx-a11y/no-redundant-roles */

import { css } from '@emotion/react';
import { Theme } from '@sumup/design-tokens';
import { TrackingElement } from '@sumup/collector';

import styled, { StyleProps } from '../../../../styles/styled';
import { shadow, hideScrollbar } from '../../../../styles/style-mixins';
import { useFocusList } from '../../../../hooks/useFocusList';
import { TOP_NAVIGATION_HEIGHT } from '../../../TopNavigation/TopNavigation';
import Headline from '../../../Headline';
import { Skeleton, SkeletonContainer } from '../../../Skeleton';
import { PrimaryLinkProps } from '../../types';
import { TRACKING_ELEMENTS } from '../../constants';
import { SecondaryLinks } from '../SecondaryLinks';
import { PrimaryLink } from '../PrimaryLink';

export interface DesktopNavigationProps {
  /**
   * Whether the navigation data is loading.
   */
  isLoading?: boolean;
  /**
   * A collection of links with nested secondary groups.
   */
  primaryLinks: PrimaryLinkProps[];
  /**
   * Text label for the primary navigation for screen readers.
   * Important for accessibility.
   */
  primaryNavigationLabel: string;
  /**
   * Text label for the secondary navigation for screen readers.
   * Important for accessibility.
   */
  secondaryNavigationLabel: string;
}

const PRIMARY_NAVIGATION_WIDTH = '48px';
const PRIMARY_NAVIGATION_OPENED_WIDTH = '220px';
const LARGE_SCREEN_BREAKPOINT = '1920px'; // max breakpoint in circuit-ui is 1280px therefore we decided to hardcode for now

const wrapperStyles = ({ theme }: StyleProps) => css`
  ${theme.mq.untilTera} {
    display: none;
  }
  ${theme.mq.tera} {
    min-width: ${PRIMARY_NAVIGATION_WIDTH};
    flex-shrink: 0;
  }
`;

const Wrapper = styled(SkeletonContainer)(wrapperStyles);

const primaryWrapperStyles = ({ theme }: StyleProps) => css`
  position: fixed;
  z-index: ${theme.zIndex.navigation};
  top: ${TOP_NAVIGATION_HEIGHT};
  bottom: 0;
  left: 0;
  height: calc(100vh - ${TOP_NAVIGATION_HEIGHT});
  width: ${theme.iconSizes.tera};
  display: flex;
  flex-direction: column;
  background-color: var(--cui-bg-normal);
  padding-top: ${theme.spacings.kilo};
  overflow-y: auto;
  overflow-x: hidden;
  box-shadow: 1px 0 var(--cui-border-divider);
  transition: width ${theme.transitions.default},
    box-shadow ${theme.transitions.default};

  &:hover,
  &:focus-within {
    ${shadow(theme)};
    width: ${PRIMARY_NAVIGATION_OPENED_WIDTH};
  }

  @media only screen and (min-width: ${LARGE_SCREEN_BREAKPOINT}) {
    width: ${PRIMARY_NAVIGATION_OPENED_WIDTH};
    &:hover,
    &:focus-within {
      box-shadow: 1px 0 var(--cui-border-divider);
    }
  }
`;

const PrimaryNavigationWrapper = styled.nav(
  primaryWrapperStyles,
  hideScrollbar,
);

const secondaryWrapperStyles = ({ theme }: StyleProps) => css`
  position: sticky;
  top: ${TOP_NAVIGATION_HEIGHT};
  display: flex;
  flex-direction: column;
  overflow-y: auto;
  background-color: var(--cui-bg-normal);
  margin-left: ${PRIMARY_NAVIGATION_WIDTH};
  height: calc(100vh - ${TOP_NAVIGATION_HEIGHT});
  width: 200px;
  border-right: ${theme.borderWidth.kilo} solid var(--cui-border-divider);
  @media only screen and (min-width: ${LARGE_SCREEN_BREAKPOINT}) {
    margin-left: ${PRIMARY_NAVIGATION_OPENED_WIDTH};
  }
`;

const SecondaryNavigationWrapper = styled.nav(secondaryWrapperStyles);

const listStyles = css`
  list-style: none;
`;

const headlineStyles = (theme: Theme) => css`
  margin: ${theme.spacings.giga} ${theme.spacings.mega} ${theme.spacings.kilo};
`;

export function DesktopNavigation({
  isLoading,
  primaryLinks,
  primaryNavigationLabel,
  secondaryNavigationLabel,
  ...props
}: DesktopNavigationProps): JSX.Element {
  const focusProps = useFocusList();

  const activePrimaryLink = primaryLinks.find((link) => link.isActive);
  const secondaryGroups =
    activePrimaryLink && activePrimaryLink.secondaryGroups;

  return (
    <TrackingElement name={TRACKING_ELEMENTS.SIDE_NAVIGATION}>
      <Wrapper isLoading={Boolean(isLoading)}>
        <PrimaryNavigationWrapper
          {...props}
          aria-label={primaryNavigationLabel}
        >
          <ul role="list" css={listStyles}>
            {primaryLinks.map((link) => (
              <li key={link.label}>
                <PrimaryLink {...link} {...focusProps} />
              </li>
            ))}
          </ul>
        </PrimaryNavigationWrapper>
        {secondaryGroups && secondaryGroups.length > 0 && (
          <SecondaryNavigationWrapper
            {...props}
            aria-label={secondaryNavigationLabel}
          >
            <Skeleton css={headlineStyles}>
              <Headline as="h2" size="four">
                {activePrimaryLink && activePrimaryLink.label}
              </Headline>
            </Skeleton>
            <SecondaryLinks
              secondaryGroups={secondaryGroups}
              trackingLabel={
                activePrimaryLink.tracking && activePrimaryLink.tracking.label
              }
            />
          </SecondaryNavigationWrapper>
        )}
      </Wrapper>
    </TrackingElement>
  );
}
