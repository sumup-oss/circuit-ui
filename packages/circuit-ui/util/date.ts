/**
 * Copyright 2024, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Temporal } from 'temporal-polyfill';
import { formatDateTime } from '@sumup-oss/intl';

import type { Locale } from './i18n.js';

export type FirstDayOfWeek = 1 | 7;
export type DaysInWeek = number;
export type PlainDateRange =
  | []
  | [Temporal.PlainDate]
  | [Temporal.PlainDate, Temporal.PlainDate];

// ISO 8601 timestamps only support positive 4-digit years
export const MIN_YEAR = 1;
export const MAX_YEAR = 9999;

export const MIN_MONTH = 1;
export const MAX_MONTH = 12;

export const MIN_DAY = 1;
// MAX_DAY is not constant as it depends on the year and month

export function getTodaysDate() {
  return Temporal.Now.plainDateISO();
}

export function isPlainDate(date: unknown): date is Temporal.PlainDate {
  return date instanceof Temporal.PlainDate;
}

export function toPlainDate(date?: string): Temporal.PlainDate | undefined {
  if (!date) {
    return undefined;
  }
  try {
    return Temporal.PlainDate.from(date);
  } catch {
    return undefined;
  }
}

export function sortDateRange<T extends PlainDateRange>(dateRange: T): T {
  return dateRange.sort((a, b) => Temporal.PlainDate.compare(a, b)) as T;
}

export function clampDate(
  date: Temporal.PlainDate,
  minDate?: Temporal.PlainDate | null,
  maxDate?: Temporal.PlainDate | null,
): Temporal.PlainDate {
  if (minDate && Temporal.PlainDate.compare(date, minDate) < 0) {
    return minDate;
  }
  if (maxDate && Temporal.PlainDate.compare(date, maxDate) > 0) {
    return maxDate;
  }
  return date;
}

export function getFirstDateOfWeek(
  date: Temporal.PlainDate,
  firstDayOfWeek: FirstDayOfWeek,
): Temporal.PlainDate {
  const diff =
    (date.dayOfWeek < firstDayOfWeek ? date.daysInWeek : 0) +
    date.dayOfWeek -
    firstDayOfWeek;
  return date.subtract({ days: diff });
}

export function getLastDateOfWeek(
  date: Temporal.PlainDate,
  firstDayOfWeek: FirstDayOfWeek,
): Temporal.PlainDate {
  return getFirstDateOfWeek(date, firstDayOfWeek).add({
    days: date.daysInWeek - 1,
  });
}

export function getMonthName(month: number, locale?: Locale) {
  // The year can be arbitrary since the month names are the same every year
  const yearMonth = new Temporal.PlainYearMonth(2000, month, 'gregory');
  return formatDateTime(yearMonth, locale, { month: 'long' });
}
