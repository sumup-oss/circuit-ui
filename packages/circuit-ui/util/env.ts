/**
 * Copyright 2024, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export function getEnvVariable(name: string) {
  if (name === 'NODE_ENV') {
    // Some bundlers have special logic for `process.env.NODE_ENV` which
    // relies on it being written as a continuous string. Destructuring or
    // dynamically accessing `NODE_ENV` can break this logic.
    throw new Error('Do not dynamically access NODE_ENV');
  }

  if (typeof process !== 'undefined') {
    return process.env?.[name];
  }

  return undefined;
}
