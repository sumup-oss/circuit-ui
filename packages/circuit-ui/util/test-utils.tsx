/**
 * Copyright 2020, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { FunctionComponent, ReactElement } from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import '@testing-library/jest-dom/extend-expect';
import { configureAxe } from 'jest-axe';
import { render as renderTest, RenderResult } from '@testing-library/react';
import { renderHook, act as actHook } from '@testing-library/react-hooks';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'emotion-theming';
import { light } from '@sumup/design-tokens';

import {
  ComponentsContext,
  defaultComponents,
} from '../components/ComponentsContext/ComponentsContext';

export * from '@testing-library/react';

export type RenderFn<T = any> = (component: ReactElement, ...rest: any) => T;

const WithProviders: FunctionComponent = ({ children }) => (
  <ComponentsContext.Provider value={defaultComponents}>
    <ThemeProvider theme={light}>{children}</ThemeProvider>
  </ComponentsContext.Provider>
);

const render: RenderFn<RenderResult> = (component, options) =>
  renderTest(component, { wrapper: WithProviders, ...options });
const renderToHtml: RenderFn<string> = (component) =>
  renderToStaticMarkup(<WithProviders>{component}</WithProviders>);
const create = (...args: Parameters<RenderFn<RenderResult>>) => {
  const { container } = render(...args);
  return container.children.length > 1
    ? container.children
    : container.firstChild;
};

const axe = configureAxe({
  rules: {
    // disabled landmark rules when testing isolated components.
    region: { enabled: false },
  },
});

export { create, render, renderToHtml, renderHook, actHook, userEvent, axe };
