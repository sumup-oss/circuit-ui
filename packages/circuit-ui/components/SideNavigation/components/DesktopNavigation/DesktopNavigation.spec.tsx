/**
 * Copyright 2021, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* eslint-disable react/display-name */
import { Home, Shop } from '@sumup/icons';

import { render, axe, RenderFn } from '../../../../util/test-utils';

import { DesktopNavigation, DesktopNavigationProps } from './DesktopNavigation';

describe('DesktopNavigation', () => {
  function renderDesktopNavigation<T>(
    renderFn: RenderFn<T>,
    props: DesktopNavigationProps,
  ) {
    return renderFn(<DesktopNavigation {...props} />);
  }

  const baseProps = {
    primaryNavigationLabel: 'Primary',
    secondaryNavigationLabel: 'Secondary',
  };

  const defaultProps = {
    ...baseProps,
    primaryLinks: [
      {
        icon: (iconProps) => <Shop {...iconProps} size="24" />,
        label: 'Shop',
        href: '/shop',
        onClick: jest.fn(),
        isActive: true,
        secondaryGroups: [
          {
            label: 'For Kids',
            secondaryLinks: [
              {
                label: 'Toys',
                href: '/shop/toys',
                onClick: jest.fn(),
              },
              {
                label: 'Books',
                href: '/shop/books',
                onClick: jest.fn(),
              },
            ],
          },
        ],
      },
    ],
  };

  describe('styles', () => {
    it('should render with secondary links', () => {
      const { container, getAllByRole } = renderDesktopNavigation(
        render,
        defaultProps,
      );

      const lists = getAllByRole('list');

      expect(lists).toHaveLength(3);
      expect(container).toMatchSnapshot();
    });

    it('should render without secondary links', () => {
      const props = {
        ...baseProps,
        primaryLinks: [
          {
            icon: (iconProps) => <Home {...iconProps} size="24" />,
            label: 'Home',
            href: '/',
            onClick: jest.fn(),
            secondaryGroups: [],
          },
        ],
      };
      const { container, getAllByRole } = renderDesktopNavigation(
        render,
        props,
      );

      const lists = getAllByRole('list');

      expect(lists).toHaveLength(1);
      expect(container).toMatchSnapshot();
    });
  });

  describe('accessibility', () => {
    it('should meet accessibility guidelines', async () => {
      const { container } = renderDesktopNavigation(render, defaultProps);
      const actual = await axe(container);
      expect(actual).toHaveNoViolations();
    });
  });
});
