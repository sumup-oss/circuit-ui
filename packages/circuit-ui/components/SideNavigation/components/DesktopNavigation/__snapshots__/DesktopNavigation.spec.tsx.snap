// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopNavigation styles should render with secondary links 1`] = `
.circuit-0[aria-busy='true'] {
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

@media (max-width: 1279px) {
  .circuit-0 {
    display: none;
  }
}

@media (min-width: 1280px) {
  .circuit-0 {
    min-width: 48px;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
  }
}

.circuit-1 {
  position: fixed;
  z-index: 800;
  top: 57px;
  bottom: 0;
  left: 0;
  height: calc(100vh - 57px);
  width: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: var(--cui-bg-normal);
  padding-top: 12px;
  overflow-y: auto;
  overflow-x: hidden;
  box-shadow: 1px 0 var(--cui-border-divider);
  -webkit-transition: width 120ms ease-in-out,box-shadow 120ms ease-in-out;
  transition: width 120ms ease-in-out,box-shadow 120ms ease-in-out;
  -ms-overflow-style: none;
  scrollbar-width: none;
}

.circuit-1:hover,
.circuit-1:focus-within {
  box-shadow: 0 3px 8px 0 rgba(0, 0, 0, 0.2);
  width: 220px;
}

@media only screen and (min-width: 1920px) {
  .circuit-1 {
    width: 220px;
  }

  .circuit-1:hover,
  .circuit-1:focus-within {
    box-shadow: 1px 0 var(--cui-border-divider);
  }
}

.circuit-1::-webkit-scrollbar {
  display: none;
}

.circuit-2 {
  list-style: none;
}

.circuit-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 80px;
  width: 100%;
  background: none;
  border: none;
  outline: none;
  text-align: left;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--cui-fg-normal);
  padding: 24px;
  -webkit-transition: color 120ms ease-in-out,background-color 120ms ease-in-out;
  transition: color 120ms ease-in-out,background-color 120ms ease-in-out;
  color: var(--cui-fg-accent);
}

.circuit-3:focus {
  outline: 0;
  box-shadow: inset 0 0 0 4px var(--cui-border-focus);
}

.circuit-3:focus::-moz-focus-inner {
  border: 0;
}

.circuit-3:focus:not(:focus-visible) {
  box-shadow: none;
}

.circuit-3:hover {
  background-color: var(--cui-bg-normal-hovered);
  color: var(--cui-fg-normal-hovered);
}

.circuit-3:active {
  background-color: var(--cui-bg-normal-pressed);
  color: var(--cui-fg-normal-pressed);
}

.circuit-3:disabled,
.circuit-3[disabled] {
  pointer-events: none;
  background-color: var(--cui-bg-normal-disabled);
  color: var(--cui-fg-normal-disabled);
}

@media (max-width: 1279px) {
  .circuit-3 {
    margin-bottom: 1px;
  }

  .circuit-3::after {
    content: '';
    display: block;
    position: absolute;
    top: 100%;
    right: 24px;
    left: 24px;
    width: calc(100% - 2 * 24px);
    border-bottom: 1px solid var(--cui-border-divider);
    -webkit-transition: width 120ms ease-in-out,right 120ms ease-in-out,left 120ms ease-in-out;
    transition: width 120ms ease-in-out,right 120ms ease-in-out,left 120ms ease-in-out;
  }
}

@media (min-width: 1280px) {
  .circuit-3 {
    height: 48px;
    width: 220px;
    padding: 12px;
    margin-bottom: 12px;
  }
}

.circuit-3:hover {
  background-color: var(--cui-bg-accent-hovered);
}

.circuit-3:active {
  background-color: var(--cui-bg-accent-pressed);
}

.circuit-4 {
  display: inline-block;
  line-height: 0;
  position: relative;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  margin-right: 12px;
}

.circuit-5 {
  display: inline-block;
  line-height: 0;
}

.circuit-6 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 700;
}

@media (max-width: 1279px) {
  .circuit-6 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

.circuit-7 {
  position: -webkit-sticky;
  position: sticky;
  top: 57px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow-y: auto;
  background-color: var(--cui-bg-normal);
  margin-left: 48px;
  height: calc(100vh - 57px);
  width: 200px;
  border-right: 1px solid var(--cui-border-divider);
}

@media only screen and (min-width: 1920px) {
  .circuit-7 {
    margin-left: 220px;
  }
}

.circuit-8 {
  display: inline-block;
  line-height: 0;
  margin: 24px 16px 12px;
}

.circuit-9 {
  font-weight: 700;
  color: var(--cui-fg-normal);
  letter-spacing: -0.03em;
  font-size: 1.125rem;
  line-height: 1.5rem;
}

.circuit-11 {
  display: inline-block;
  line-height: 0;
  margin: 32px 16px 8px;
}

.circuit-12 {
  text-transform: uppercase;
  font-weight: 700;
  font-size: 0.875rem;
  line-height: 1.25rem;
  color: var(--cui-fg-normal);
}

.circuit-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  color: var(--cui-fg-normal);
  background-color: var(--cui-bg-normal);
  text-align: left;
  cursor: pointer;
  -webkit-transition: color 120ms ease-in-out,background-color 120ms ease-in-out;
  transition: color 120ms ease-in-out,background-color 120ms ease-in-out;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 16px 24px;
  word-break: break-word;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
}

.circuit-14:hover {
  background-color: var(--cui-bg-normal-hovered);
  color: var(--cui-fg-normal-hovered);
}

.circuit-14:active {
  background-color: var(--cui-bg-normal-pressed);
  color: var(--cui-fg-normal-pressed);
}

.circuit-14:focus {
  outline: 0;
  box-shadow: inset 0 0 0 4px var(--cui-border-focus);
}

.circuit-14:focus::-moz-focus-inner {
  border: 0;
}

.circuit-14:focus:not(:focus-visible) {
  box-shadow: none;
}

.circuit-14:disabled,
.circuit-14[disabled] {
  pointer-events: none;
  background-color: var(--cui-bg-normal-disabled);
  color: var(--cui-fg-normal-disabled);
}

@media (min-width: 1280px) {
  .circuit-14 {
    padding: 12px;
    padding-left: 20px;
  }
}

.circuit-15 {
  display: inline-block;
  line-height: 0;
  margin-right: 8px;
}

.circuit-16 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
}

<div>
  <div
    aria-busy="false"
    class="circuit-0"
  >
    <nav
      aria-label="Primary"
      class="circuit-1"
    >
      <ul
        class="circuit-2"
        role="list"
      >
        <li>
          <a
            aria-current="page"
            class="circuit-3"
            data-focus-list="1"
            href="/shop"
          >
            <span
              class="circuit-4"
            >
              <svg
                fill="none"
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7 23a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm14-3a3 3 0 1 0-5.999 0A3 3 0 0 0 21 20zm0-16H5.114a2 2 0 0 1-1.921-1.445L2.92 1.61a1 1 0 0 0-1.9.59l3.24 11.35A2 2 0 0 0 6.18 15H19a2 2 0 0 0 1.92-1.45l2-7c.052-.179.08-.364.08-.55a2.006 2.006 0 0 0-2-2z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="circuit-5"
            >
              <span
                class="circuit-6"
              >
                Shop
              </span>
            </span>
          </a>
        </li>
      </ul>
    </nav>
    <nav
      aria-label="Secondary"
      class="circuit-7"
    >
      <span
        class="circuit-8"
      >
        <h2
          class="circuit-9"
        >
          Shop
        </h2>
      </span>
      <ul
        class="circuit-2"
        role="list"
      >
        <li>
          <span
            class="circuit-11"
          >
            <h3
              class="circuit-12"
            >
              For Kids
            </h3>
          </span>
          <ul
            class="circuit-2"
            role="list"
          >
            <li>
              <a
                class="circuit-14"
                data-focus-list="2"
                href="/shop/toys"
              >
                <span
                  class="circuit-15"
                >
                  <p
                    class="circuit-16"
                  >
                    Toys
                  </p>
                </span>
              </a>
            </li>
            <li>
              <a
                class="circuit-14"
                data-focus-list="2"
                href="/shop/books"
              >
                <span
                  class="circuit-15"
                >
                  <p
                    class="circuit-16"
                  >
                    Books
                  </p>
                </span>
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`DesktopNavigation styles should render without secondary links 1`] = `
.circuit-0[aria-busy='true'] {
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

@media (max-width: 1279px) {
  .circuit-0 {
    display: none;
  }
}

@media (min-width: 1280px) {
  .circuit-0 {
    min-width: 48px;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
  }
}

.circuit-1 {
  position: fixed;
  z-index: 800;
  top: 57px;
  bottom: 0;
  left: 0;
  height: calc(100vh - 57px);
  width: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: var(--cui-bg-normal);
  padding-top: 12px;
  overflow-y: auto;
  overflow-x: hidden;
  box-shadow: 1px 0 var(--cui-border-divider);
  -webkit-transition: width 120ms ease-in-out,box-shadow 120ms ease-in-out;
  transition: width 120ms ease-in-out,box-shadow 120ms ease-in-out;
  -ms-overflow-style: none;
  scrollbar-width: none;
}

.circuit-1:hover,
.circuit-1:focus-within {
  box-shadow: 0 3px 8px 0 rgba(0, 0, 0, 0.2);
  width: 220px;
}

@media only screen and (min-width: 1920px) {
  .circuit-1 {
    width: 220px;
  }

  .circuit-1:hover,
  .circuit-1:focus-within {
    box-shadow: 1px 0 var(--cui-border-divider);
  }
}

.circuit-1::-webkit-scrollbar {
  display: none;
}

.circuit-2 {
  list-style: none;
}

.circuit-3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 80px;
  width: 100%;
  background: none;
  border: none;
  outline: none;
  text-align: left;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--cui-fg-normal);
  padding: 24px;
  -webkit-transition: color 120ms ease-in-out,background-color 120ms ease-in-out;
  transition: color 120ms ease-in-out,background-color 120ms ease-in-out;
}

.circuit-3:focus {
  outline: 0;
  box-shadow: inset 0 0 0 4px var(--cui-border-focus);
}

.circuit-3:focus::-moz-focus-inner {
  border: 0;
}

.circuit-3:focus:not(:focus-visible) {
  box-shadow: none;
}

.circuit-3:hover {
  background-color: var(--cui-bg-normal-hovered);
  color: var(--cui-fg-normal-hovered);
}

.circuit-3:active {
  background-color: var(--cui-bg-normal-pressed);
  color: var(--cui-fg-normal-pressed);
}

.circuit-3:disabled,
.circuit-3[disabled] {
  pointer-events: none;
  background-color: var(--cui-bg-normal-disabled);
  color: var(--cui-fg-normal-disabled);
}

@media (max-width: 1279px) {
  .circuit-3 {
    margin-bottom: 1px;
  }

  .circuit-3::after {
    content: '';
    display: block;
    position: absolute;
    top: 100%;
    right: 24px;
    left: 24px;
    width: calc(100% - 2 * 24px);
    border-bottom: 1px solid var(--cui-border-divider);
    -webkit-transition: width 120ms ease-in-out,right 120ms ease-in-out,left 120ms ease-in-out;
    transition: width 120ms ease-in-out,right 120ms ease-in-out,left 120ms ease-in-out;
  }
}

@media (min-width: 1280px) {
  .circuit-3 {
    height: 48px;
    width: 220px;
    padding: 12px;
    margin-bottom: 12px;
  }
}

.circuit-4 {
  display: inline-block;
  line-height: 0;
  position: relative;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  margin-right: 12px;
}

.circuit-5 {
  display: inline-block;
  line-height: 0;
}

.circuit-6 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
}

@media (max-width: 1279px) {
  .circuit-6 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

<div>
  <div
    aria-busy="false"
    class="circuit-0"
  >
    <nav
      aria-label="Primary"
      class="circuit-1"
    >
      <ul
        class="circuit-2"
        role="list"
      >
        <li>
          <a
            class="circuit-3"
            data-focus-list="3"
            href="/"
          >
            <span
              class="circuit-4"
            >
              <svg
                fill="none"
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 9a1 1 0 0 1-1.57.82l-1.43-1V20.5a2.5 2.5 0 0 1-2.5 2.5H15a1 1 0 0 1-1-1v-5a1 1 0 0 0-1-1h-2a1 1 0 0 0-1 1v5a1 1 0 0 1-1 1H6.5A2.5 2.5 0 0 1 4 20.5V8.82l-1.43 1A1.011 1.011 0 0 1 2 10a1 1 0 0 1-.57-1.82l10-7a.992.992 0 0 1 1.14 0l10 7A.999.999 0 0 1 23 9z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <span
              class="circuit-5"
            >
              <span
                class="circuit-6"
              >
                Home
              </span>
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
